stream/src/placeholder/utils/time.rs

```diff
@@ -0,0 +1,21 @@
+use std::time::Duration;
+
+/// Convert a `Duration` to epoch milliseconds.
+/// If the duration is too large, it will be clamped to the maximum value of `u64`.
+pub fn to_millis(duration: Duration) -> u64 {
+    duration.as_secs()
```

Comment:
nit: An alternative approach to this would be:
```
duration.as_millis().min(u64::MAX as u128) as u64
```

Not sure what I like better (both are correct) but figured I'd share in any case.